src/sortStudents.ts

```diff
@@ -1,16 +1,68 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyStudents = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  function calculateAverage(grades: number[]):number {
+    return grades.reduce((sum, value) => sum + value, 0) / grades.length;
+  }
+
+  copyStudents.sort((a: Student, b: Student) => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        if (order === 'asc') {
+          return a[sortBy].localeCompare(b[sortBy]);
+        }
+
+        return b[sortBy].localeCompare(a[sortBy]);
```

Comment:
```suggestion
        return a[sortBy].localeCompare(b[sortBy]) * direction;
```

You can avoid a lot of code duplication by using variable `direction` as mulitplier: if order is 'asc' it will be 1, else it will be -1.